R/GeospatialFunctions.R

```diff
@@ -498,216 +961,523 @@ TADA_GetATTAINS <- function(.data, return_sf = TRUE) {
+          Visit_Count = length(unique(ActivityStartDate)),
+          Parameter_Count = length(unique(CharacteristicName)),
+          Organization_Count = length(unique(OrganizationIdentifier)),
+          ATTAINS_AUs = as.character(list(unique(ATTAINS.assessmentunitidentifier)))
+        ) %>%
+        dplyr::mutate(
+          ATTAINS_AUs = ifelse(is.na(ATTAINS_AUs), "None", ATTAINS_AUs),
+          LatitudeMeasure = as.numeric(LatitudeMeasure),
+          LongitudeMeasure = as.numeric(LongitudeMeasure)
         )
-      ) %>%
-      leaflet::clearShapes() %>%
-      leaflet::fitBounds(
-        lng1 = min(sumdat$LongitudeMeasure),
-        lat1 = min(sumdat$LatitudeMeasure),
-        lng2 = max(sumdat$LongitudeMeasure),
-        lat2 = max(sumdat$LatitudeMeasure)
-      ) %>%
-      leaflet.extras::addResetMapButton() %>%
-      leaflet::addLegend(
-        position = "bottomright",
-        colors = c(tada.pal[3], tada.pal[4], tada.pal[7], "black", NA),
-        labels = c(
-          "ATTAINS: Not Supporting", "ATTAINS: Supporting", "ATTAINS: Not Assessed", "Water Quality Observation(s)",
-          "NHDPlus HR catchments containing water quality observations + ATTAINS feature are represented as clear polygons with black outlines."
-        ),
-        opacity = 1,
-        title = "Legend"
+      
+      # Basemap for AOI:
+      map <- leaflet::leaflet() %>%
+        leaflet::addProviderTiles("Esri.WorldTopoMap",
+                                  group = "World topo",
+                                  options = leaflet::providerTileOptions(
+                                    updateWhenZooming = FALSE,
+                                    updateWhenIdle = TRUE
+                                  )
+        ) %>%
+        leaflet::clearShapes() %>%
+        leaflet::fitBounds(
+          lng1 = min(sumdat$LongitudeMeasure),
+          lat1 = min(sumdat$LatitudeMeasure),
+          lng2 = max(sumdat$LongitudeMeasure),
+          lat2 = max(sumdat$LatitudeMeasure)
+        ) %>%
+        leaflet.extras::addResetMapButton() %>%
+        leaflet::addLegend(
+          position = "bottomright",
+          colors = c(tada.pal[3], tada.pal[4], tada.pal[7], "black", NA),
+          labels = c(
+            "ATTAINS: Not Supporting", "ATTAINS: Supporting", "ATTAINS: Not Assessed", "Water Quality Observation(s)",
+            "NHDPlus HR catchments containing water quality observations + ATTAINS feature are represented as clear polygons with black outlines."
+          ),
+          opacity = 1,
+          title = "Legend"
+        )
+      
+      # Add ATTAINS catchment outlines (if they exist):
+      try(
+        map <- map %>%
+          leaflet::addPolygons(
+            data = ATTAINS_catchments,
+            color = "black",
+            weight = 1, fillOpacity = 0,
+            popup = paste0("NHDPlus HR Catchment ID: ", ATTAINS_catchments$nhdplusid)
+          ),
+        silent = TRUE
```

Comment:
I think (?) #492  I was going to submit PR for would potentially leverage some of this additional code and code that follows?  In any event, anything I submit for this issue I can make sure dovetails with anything in this PR